flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -461,6 +463,23 @@ public boolean dropTemporaryFunction(String path) {
         return functionCatalog.dropTemporaryCatalogFunction(unresolvedIdentifier, true);
     }
 
+    @Override
+    public void addJar(String jarPath) {
+        try {
+            resourceManager.registerJarResources(
+                    Collections.singletonList(new ResourceUri(ResourceType.JAR, jarPath)));
+        } catch (IOException e) {
+            throw new TableException(String.format("Could not register the specified jar [%s].", jarPath), e);
+        }
+    }
+
+    @Override
+    public List<String> listJars() {
```

Comment:
AddJar and listJars could work for DataStream user or TableAPI user.
How to use this function for SQL user?
We also need to support `AddJarOperation` in `executeInternal` method, WDYT?

Reply:
As @lsyldliu, the public api will be discussed later.